Explorer/Assets/DCL/UI/ConnectionStatusPanel/ConnectionStatusPanelController.cs

```diff
@@ -0,0 +1,146 @@
+using DCL.Utilities;
+using ECS.SceneLifeCycle;
+using ECS.SceneLifeCycle.CurrentScene;
+using LiveKit.Proto;
+using MVC;
+using System;
+using System.Threading;
+
+namespace DCL.UI.ConnectionStatusPanel
+{
+    public partial class ConnectionStatusPanelController : ControllerBase<ConnectionStatusPanelView>
+    {
+        private readonly IUserInAppInitializationFlow userInAppInitializationFlow;
+        private readonly IMVCManager mvcManager;
+        private readonly ICurrentSceneInfo currentSceneInfo;
+        private readonly ECSReloadScene ecsReloadScene;
+        private readonly IRoomsStatus roomsStatus;
+        private readonly World world;
+        private readonly Entity playerEntity;
+        private readonly CancellationTokenSource cancellationTokenSource = new ();
+        private bool isSceneReloading;
+
+        public override CanvasOrdering.SortingLayer Layer => CanvasOrdering.SortingLayer.Persistent;
+
+        public ConnectionStatusPanelController(
+            ViewFactoryMethod viewFactory,
+            IUserInAppInitializationFlow userInAppInitializationFlow,
+            IMVCManager mvcManager,
+            ICurrentSceneInfo currentSceneInfo,
+            ECSReloadScene ecsReloadScene,
+            IRoomsStatus roomsStatus,
+            World world,
+            Entity playerEntity
+        ) : base(viewFactory)
+        {
+            this.userInAppInitializationFlow = userInAppInitializationFlow;
+            this.mvcManager = mvcManager;
+            this.currentSceneInfo = currentSceneInfo;
+            this.ecsReloadScene = ecsReloadScene;
+            this.roomsStatus = roomsStatus;
+            this.world = world;
+            this.playerEntity = playerEntity;
+        }
+
+        protected override void OnViewInstantiated()
+        {
+            currentSceneInfo.SceneStatus.OnUpdate += SceneStatusOnUpdate;
+            SceneStatusOnUpdate(currentSceneInfo.SceneStatus.Value);
+            Bind(roomsStatus.ConnectionQualityScene, viewInstance.SceneRoom);
+            Bind(roomsStatus.ConnectionQualityIsland, viewInstance.GlobalRoom);
+        }
+
+        private void SceneStatusOnUpdate(ICurrentSceneInfo.Status? obj)
+        {
+            const float DELAY = 5f;
+
+            async UniTaskVoid ShowButtonAsync()
+            {
+                await UniTask.Delay(TimeSpan.FromSeconds(DELAY));
+                viewInstance.Scene.ShowReloadButton(TryReloadScene);
+            }
+
+            if (obj is not { } status)
+            {
+                viewInstance.Scene.HideStatus();
+                return;
+            }
+
+            viewInstance.Scene.ShowStatus(status);
+
+            if (status is ICurrentSceneInfo.Status.Crashed)
+                ShowButtonAsync().Forget();
+        }
+
+        private void Bind(IReadonlyReactiveProperty<ConnectionQuality> value, IStatusEntry statusEntry)
+        {
+            value.OnUpdate += newValue => UpdateStatusEntry(statusEntry, newValue);
```

Comment:
Missing unsubscribe to avoid possible issues. Also, will have to refactor into a non-anonym function to properly unsubscribe